src/main/java/com/minecolonies/entity/ai/util/ChatSpamFilter.java

```diff
@@ -86,41 +74,31 @@ public void requestWithoutSpam(@NotNull String chat, int ticks)
      * Helpful for requesting items.
      *
      * @param key   the translation key
-     * @param ticks the amount of ticks waited
      * @param chat  the chat message
      */
-    public void talkWithoutSpam(String key, int ticks, String... chat)
+    public void talkWithoutSpam(String key, String... chat)
     {
         String curstring = key + Arrays.toString(chat);
         if (Objects.equals(speechDelayString, curstring))
         {
-            if (speechDelay > 0)
+            if (speechDelay > worker.getOffsetTicks())
             {
-                speechDelay -= ticks;
                 return;
             }
-            speechRepeat++;
+
+            // to protect against overflows
+            if ((BASE_TIMEOUT << speechRepeat) < MAX_TIMEOUT)
```

Comment:
Can you add a comment explaining how this bitshift works and what it does exactly? That would greatly improve maintainability.